android/src/main/java/com/marigold/rnsdk/RNMessageStreamModule.kt

```diff
@@ -27,20 +30,56 @@ class RNMessageStreamModule (reactContext: ReactApplicationContext, private val
     @VisibleForTesting
     internal var jsonConverter: JsonConverter = JsonConverter()
 
+    private val eventChannel = Channel<Boolean>()
+    private var defaultInAppNotification = displayInAppNotifications
     init {
         messageStream.setOnInAppNotificationDisplayListener(this)
     }
 
     override fun shouldPresentInAppNotification(message: Message): Boolean {
-        try {
-            val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
-            reactApplicationContext
+        if (defaultInAppNotification) {
+            return true
+        }
+
+        return runBlocking {
+            val result = async { emitWithTimeout(message) }
+            result.await()
+        }
+    }
+
+    private suspend fun emitWithTimeout(message: Message): Boolean {
+        return withTimeoutOrNull(5000L) {
+            try {
+                val writableMap = jsonConverter.convertJsonToMap(message.toJSON())
+                val emitter = reactApplicationContext
                     .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter::class.java)
-                    .emit("inappnotification", writableMap)
-        } catch (e: JSONException) {
-            e.printStackTrace()
+                emitter.emit("inappnotification", writableMap)
+                eventChannel.receive()
+            } catch (e: JSONException) {
+                e.printStackTrace()
+                true
+            }
+        } ?: true
+    }
+
+    @ReactMethod
+    fun acknowledgeEvent() {
```

Comment:
Can we add a boolean to this method? This will allow the RN side to implement its own logic on whether it should handle the in-app notification, then respond with the result.